src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
Do you mean something like `CSharpCodeWriter WriteHtmlToken(ICollection<RazorIRNode> nodes)` in the `CSharpCodeWriter` class?
In my opinion `RazorIRNode` should not mixed with the `CSharpCodeWriter` that is at a lower level... But at the end it's up to you 😄